cloudsync_onedrive.py

```diff
@@ -386,16 +387,20 @@ def _direct_api(self, action, path=None, *, url=None, stream=None, data=None, he
 
         return res
 
-    def _safe_direct_api(self, path, method="get", default=None):
+    def _direct_api_error_trap(self, path, method="get", default=None):
         # wrapper for _direct_api that raises only connectivity errors - other errors are caught and logged
         try:
             return self._direct_api(method, path)
         except (CloudDisconnectedError, CloudTokenError, CloudTemporaryError):
             raise
         except Exception as e:
-            log.warning("%s failed with %s", path, repr(e))
+            self._save_namespace_error(f"{path} failed with {repr(e)}")
             return default
 
+    def _save_namespace_error(self, error: str):
+        d = Drive(id="", name=error, parent=self._namespace_errors)
+        log.warning(d.name)
```

Comment:
Easier to read:
```suggestion
        log.warning(error)
```

Reply:
did this initially to satisfy lint (unused variable d) -- changed d to _ instead